lib/iphone-simulator-xcode-simctl.ts

```diff
@@ -234,15 +234,15 @@ export class XCodeSimctlSimulator extends IPhoneSimulatorNameGetter implements I
 	}
 
 	public startSimulator(device?: IDevice): void {
-		if (!device || !device.runtimeVersion || !device.fullId) {
-			device = this.getDeviceToRun(device);
-		}
-
 		// In case the id is undefined, skip verification - we'll start default simulator.
 		if (device.id) {
 			this.verifyDevice(device);
 		}
 
+		if (!device || !device.runtimeVersion || !device.fullId) {
+			device = this.getDeviceToRun(device);
```

Comment:
What happens if we don't have any devices to run? Should we have another check here?

Reply:
Actually `getDeviceToRun` always returns a device so it is not possible device to be null here.

Reply:
If `this.simctl.getDevices()` returns empty array it might be undefined, but I agree it is unlikely.